pkg/logs/pipeline/provider_mock.go

```diff
@@ -0,0 +1,17 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+package pipeline
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/logs/message"
+)
+
+// MockPipelineChans initializes pipelinesChans for testing purpose
+func (pp *Provider) MockPipelineChans() {
```

Comment:
I'd put this in the test package, otherwise this would be part of the `Provider` api.

Reply:
If I moved this to `pipeline_test.go`, the structure won't be exported: https://stackoverflow.com/questions/22277537/export-structure-only-for-testing-in-golang
A good solution would be to create an interface with two different implementations.